app/celery/tasks.py

```diff
@@ -192,15 +195,17 @@ def process_rows(rows: List, template: Template, job: Job, service: Service):
     # the same_sms and save_email task are going to be using template and service objects from cache
     # these objects are transient and will not have relationships loaded
     if encrypted_smss:
-        save_smss.apply_async(
```

Comment:
Here we're wrapping the call to the `save_...` tasks in a chain with the `job_complete` task such that they are executed sequentially. By calling `apply_async()` on the chain itself we can preserve the async behaviour we had previously.